### PR TITLE
feat: allow to run prepare steps sequentially

### PR DESCRIPTION
**Pull request by KillianHmyd**
This can be useful in the case where we must build a whole package in the prepare step.
In my specific case we use multi-semantic-release to publish some common types and a Docker image. In the prepare step we run a `docker build` in which we need to re-install the dependencies.

Because sometimes the prepare step is not over in the type package the build result in an error because the inlinePlugin started writing half of the package.json to update the dependencies version.

### CODE REVIEW

File: lib/createInlinePluginCreator.js [hunk 1]
```diff
@@ -181,6 +181,11 @@ function createInlinePluginCreator(packages, multiContext, synchronizer, flags)
 		};
 
 		const prepare = async (pluginOptions, context) => {
+			if (pkg.options.sequentialPrepare) {
```
Comment (antongolub) [inline lib/createInlinePluginCreator.js:184]: We're going to separate **semrel** and **msr** flags in future. So let's use `flags.sequentialPrepare` here.

**Review by antongolub — CHANGES_REQUESTED** [review]
Hey, @KillianHmyd,

could you also add a test for this feature?

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
-	"name": "@qiwi/multi-semantic-release",
+	"name": "@payfit/multi-semantic-release",
 	"author": "Dave Houlbrooke <dave@shax.com>",
-	"version": "3.12.4",
+	"version": "3.14.0-rc.2",
```
Comment (antongolub) [inline package.json:4]: This should be reverted ) Missed this point.

**Review by antongolub — CHANGES_REQUESTED** [review]

**Comment by KillianHmyd** [comment]
@antongolub Sorry for the package.json, it's reverted 👍 
I've added the test and used the flag instead of the options property.

I've also added a check to be sure that no one use this option while having cyclic dependencies.

**Review by antongolub — APPROVED** [review]
LGTM

**Comment by antongolub** [comment]
@KillianHmyd,

It's on the way. Thanks for the contribution.

**Pull request merged**